NuRadioReco/modules/io/noise/noiseImporterUproot.py

```diff
@@ -0,0 +1,195 @@
+                else:
+                    bunches = list(np.arange(0, nt_uproot.num_entries, 1000))
+                    bunches.append(nt_uproot.num_entries)
+                    first = bunches[:-1]
+                    last = bunches[1:]
+                    for i in range(len(first)):
+                        logger.info("reading data bunch {} of {}".format(i+1, len(first)))
+                        data.append(np.array(nt_uproot["AmpOutData."]["AmpOutData.fData"].array(entry_start=first[i], entry_stop=last[i])) * units.mV)
+                # trigger jagged array only consists of single number, so drop the array [:,0]
+                logger.debug("reading trigger info")
+                trigger.append(np.array(nt_uproot['EventHeader.']['EventHeader.fTrgInfo'].array(interpretation = ARIANNA_uproot_interpretation['trigger']))[:,0])
+                # consists of posix time [s] and [ns]
+                logger.debug("reading times")
+                event_times_file = np.array(nt_uproot['EventHeader.']['EventHeader.fTime'].array(interpretation = ARIANNA_uproot_interpretation['time']))[:,0]
+                posix_times.append(event_times_file)
+                # read the temperature
+                logger.debug("reading temperature")                
+                (temp_times, interpolated_temperatures) = self._read_temperature_curve(nf_uproot, event_times_file)
+                temperature.append(interpolated_temperatures)
+                # read the run number
+                run.append(np.array(nt_uproot['EventMetadata./EventMetadata.fRun'].array()))
+                # read the station id
+                station_id.append(np.array(nt_uproot['EventMetadata./EventMetadata.fStnId'].array()))
+                # read DTms value
+                dtms.append(np.array(nt_uproot['EventHeader./EventHeader.fDTms'].array()))
+                # read station voltages
+                (temp_times, interpolated_voltages) = self._read_power_curve(nf_uproot, event_times_file)
+                power_voltage.append(interpolated_voltages)
+
+        self.data = np.concatenate(data)
+        self.posix_time = np.concatenate(posix_times)
+        self.datetime = np.array([np.datetime64(int(t), 's') for t in self.posix_time])
+        self.trigger = np.concatenate(trigger)
+        self.temperature = np.concatenate(temperature)
+        self.power = np.concatenate(power_voltage)
+        self.station_id = np.concatenate(station_id)
+        self.dtms = np.concatenate(dtms)
+        self.run_number = np.concatenate(run)
+        self.nevts = len(self.data)
+
+    def _read_temperature_curve(self, nf, interpolation_times=None):
+        """
+        Imports the temperature vs. posix time from a noise file (nf)
+
+        The temperature is not taken per event, so the posix_time and temperature value is returned
+        if interpolation_times is specified, interpolate to these posix time stamps
```

Comment:
This doc string doesn't quite make it clear what format the interpolation_times should have.